chpl-src/announcing-chapel-2.6.chpl

```diff
@@ -0,0 +1,463 @@
+  prototypical form for the 2.5 release, but in 2.6 it is now
+  supported by all compiler back-ends and has improved stability.
+
+  To use this feature, you must first have a _shared library_ (using
+  Linux terminology) that you wish to dynamically load.  As a simple
+  example, we'll create and call into a toy library defined as
+  follows:
+
+{{<file_download fname="MyAdd.c" lang="C">}}
+
+  The compiler invocation to create a dynamic library from the C code
+  above should look something like the following, where details may
+  vary depending on your C compiler and platform:
+
+
+  ```console
+  $ clang -shared -fPIC -o libMyAdd.so MyAdd.c
+  ```
+
+
+  Once your library is compiled, you can write a program to load it
+  and call `myAdd()` with just a few lines of Chapel code:
+
+*/
+
+ use DynamicLoading;
+
+ const lib = binary.load('./libMyAdd.so'),
+       add = lib.retrieve('myAdd', proc(x: int, y: int): int);
+
+ const n = add(2, 2);
+ writeln(n);
+
+/*
+
+  Note that although `bin.retrieve()` was only called on `Locales[0]`,
+  the retrieved procedure stored in `add` can be invoked on any
+  locale:
+
+*/
+
+on Locales.last {
+  const n = add(here.id, here.id);
+  writeln(n);
+}
+
+/*
+
+  In order to use dynamic loading in Chapel at present, the
+  `useProcedurePointers` `config param` must be set to true during
+  Chapel compilation. This requirement will be relaxed in future
+  releases.  For example, to compile this example and run it on four
+  locales, you could use:
+
+  ```console
+   $ chpl -suseProcedurePointers=true UseMyAdd.chpl
+   $ ./UseMyAdd -nl 4
+   4
+   6
+  ```
+
+  In addition to supporting traditional shared libraries like this
+  sample C library, this feature also supports loading and calling
+  into dynamic Chapel libraries whose exported routines are pure and
+  C-like—for example, ones that don't rely on Chapel's runtime or
+  modules.  In future releases, we plan to expand these features to
+  support arbitrary Chapel code.
+
+
+  ### Debugging Enhancements
+
+  Debugging Chapel programs gets even better in version 2.6 with
+  better debug information and new pretty-printers.  Historically,
+  debugging Chapel programs has meant interacting with the generated C
+  code.  This meant that when inspecting Chapel variables, you would
+  see the internal C representation of those data structures.  In this
+  release, we added pretty-printers for LLDB when using Chapel's C
+  back-end that make it possible to view Chapel data structures using
+  formats that are much more intuitive and user-oriented.  This
+  improvement can be seen through the following sample debugging
+  session involving arrays.
+
+  In this example, we'll debug the following simple Chapel program:
+
+  {{<file_download fname="example.chpl" lang="Chapel">}}
+
+  We have used the `Debugger.breakpoint` {{<sidenote "right"
+  "pseudo-statement" -5>}}This is actually a parentheses-less
```

Comment:
what makes is `pseudo`? Its a statement that calls int parenless function `breakpoint`

Reply:
Hmm, I guess I can get behind that.  I used "pseudo" since it isn't a primitive statement type or keyword in the language like `return;`, `break;`, `continue;`, etc.  But I guess `foo(a, b, c);` is a statement as well that uses no keywords, so I stand corrected.